### DISCUSSION TITLE
Express and EJS problems

### DISCUSSION THREAD

**Discussion by Milo123459** [vercel/vercel#4541] [Help] [answered]
Hiya, I recently created my own website with EJS and Express, but when I go to the domain, I see the actual code, not my website. Any ideas?

**Comment by mcsdevv** [comment, marked as answer]
Hello Milo, this usually happens you try to run a server instead of static content.

The Vercel platform is meant for static sites and Serverless Functions, it does not support running an Express server.

This article provides further information: https://vercel.com/knowledge/npm-run-start-not-working

**Reply by Milo123459** [reply]
Alright, thanks! Really appreciate it, man :)

**Reply by Milo123459** [reply]
I just read it - how do I make my express and ejs site work on vercel tho?

**Reply by NANO-ck** [reply]
> I just read it - how do I make my express and ejs site work on vercel tho?

Hi, I'm also facing that issue, how do I use ejs on vercel? 
Thank you in advance

**Reply by Milo123459** [reply]
You don't.

**Reply by NANO-ck** [reply]
wait so we can't use ejs on vercel at all??

**Reply by NANO-ck** [reply]
we can only use html then?

**Comment by KashiwalHarsh** [comment]
Hey, @NANO-ck i found this issue today,
Did you find any solution on how to work with  ejs and vercel ?

**Reply by NANO-ck** [reply]
TLDR: Don't use EJS on vercel, use HTML or Next/Vue

Hey,
I searched a lot, and there is absolutely no way to work with EJS on vercel, we need to build the pages in order for vercel to provide the content in a static manner.

The alternatives I used were using HTML, and I also learnt Next.JS which is very powerful and is more useful than EJS.

**Comment by KashiwalHarsh** [comment]
Thanks for the response @NANO-ck 
I encountered this issue today only and I want my project to be deployed urgently and EJS is a must for this project because I want to display API response to the front end. I Understand and realised that Next/Vue/React does this job very easily but is there any other alternative to host an EJS-type deployment

**Reply by NANO-ck** [reply]
Alternatively, you may insert the API response using javascript on an HTML page

**Reply by KashiwalHarsh** [reply]
Well that seems like some work : )
Thanks for your help I wasn't really expecting a reply.
Can we connect ?

**Reply by NANO-ck** [reply]
Yes we can

Converting the EJS API implementation shouldn't be too complicated if ever there is no specific security in the request

**Reply by KashiwalHarsh** [reply]
I get the idea that we can insert Dynamic data to HTML using javascript, Since I am relatively new to all this It might take a while.
